### PR TITLE
Rename cashier to deposit

### LINKED ISSUE #1660: Rename `cashier` to `deposit` in API endpoints
https://all-rebilly.redoc.ly/tag/Cashier 

Cashier implies both deposits and withdrawals. We need to rename the API endpoints and update the definitions: 
For example: `PostCashierRequest` -> `PostDepositRequest`

### PR DESCRIPTION
**Pull request by swilczynski**
## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by noon-dawg — DISMISSED** [review]
Looks good on the docs side

File: openapi/components/requestBodies/PostDepositRequest.yaml [hunk 1]
```diff
@@ -27,11 +27,11 @@ properties:
           multipleOf: 1
           maximum: 10000
       ```
-      For more information, see [Create a cashier strategy](../PostCashierStrategy).
+      For more information, see [Create a deposit strategy](https://all-rebilly.redoc.ly/tag/Deposits/#operation/PostDepositStrategy).
```
Comment (justnero) [inline openapi/components/requestBodies/PostDepositRequest.yaml:30]: Why is this an absolute link and not a relative one?

Comment (swilczynski) [reply openapi/components/requestBodies/PostDepositRequest.yaml:30]: Relative links don't work for some reason, I tried `../PostDepositStrategy`, `./tags/Deposits/#operations/PostDepositStrategy` and many more. We have many relative links in API definitions that aren't working, some absolute links that aren't working too. Only one that worked is the variation that I used. I guess we'll have to fix it one day but this PR is time pressing so I'd leave it for a followup one

File: openapi/components/requestBodies/PostDepositRequest.yaml [hunk 2]
```diff
@@ -40,8 +40,8 @@ properties:
     description: |-
       List of available deposit amounts.
 
-      If `amounts` is not specified when a cashier is created, amounts are determined from the chosen strategy.
-      For more information, see the [`strategyId` property](../PostCashierRequest#!path=strategyId&t=request).
+      If `amounts` is not specified when a deposit is created, amounts are determined from the chosen strategy.
```
Comment (colinbird) [inline openapi/components/requestBodies/PostDepositRequest.yaml:43]:
```suggestion
      If `amounts` is not specified when a deposit request is created, amounts are determined from the chosen strategy.
```

File: openapi/components/requestBodies/PostDepositRequest.yaml [hunk 3]
```diff
@@ -51,8 +51,8 @@ properties:
     description: |-
       Custom amount restrictions.
       If this value is `null`,  custom amounts are prohibited.
-      If `customAmount` is not specified when a cashier is created, amount restrictions are determined from the chosen strategy.
-      For more information, see the [`strategyId` property](../PostCashierRequest#!path=strategyId&t=request).
+      If `customAmount` is not specified when a deposit is created, amount restrictions are determined from the chosen strategy.
```
Comment (colinbird) [inline openapi/components/requestBodies/PostDepositRequest.yaml:54]:
```suggestion
      If `customAmount` is not specified when a deposit request is created, amount restrictions are determined from the chosen strategy.
```

File: openapi/components/schemas/DepositRequest.yaml [hunk 4]
```diff
@@ -112,8 +112,8 @@ properties:
   properties:
     readOnly: true
     description: |-
-      Properties that are available for the user to complete when they use the hosted cashier form.
-      Use this object to describe fields that are rendered and completed during the deposit process, in the cashier deposit form.
+      Properties that are available for the user to complete when they use the hosted deposit form.
+      Use this object to describe fields that are rendered and completed during the deposit process, in the deposit deposit form.
```
Comment (colinbird) [inline openapi/components/schemas/DepositRequest.yaml:116]:
```suggestion
      Properties that are available for the user to complete when they use the hosted deposit form.
      Use this object to describe fields that are rendered and completed on the hosted deposit form.
```

Comment (colinbird) [reply openapi/components/schemas/DepositRequest.yaml:116]: Maybe still a bit redundant

File: openapi/components/schemas/DepositStrategy.yaml [hunk 5]
```diff
@@ -63,16 +63,16 @@ properties:
...
 
           If the customer has no successful deposits, `baseAmount` is not adjusted.
   customAmount:
     type: object
     description: |-
       Custom amount restrictions.
       If this value is `null`,  custom amounts are prohibited.
-      If `customAmount` is not specified when a cashier is created, amount restrictions are determined from the chosen strategy.
-      For more information, see the [`strategyId` property](../PostCashierRequest#!path=strategyId&t=request).
+      If `customAmount` is not specified when a deposit is created, amount restrictions are determined from the chosen strategy.
```
Comment (lexerom) [inline openapi/components/schemas/DepositStrategy.yaml:74]:
```suggestion
      If `customAmount` is not specified when a deposit request is created, amount restrictions are determined from the chosen strategy.
```

**Comment by swilczynski** [comment]
@Rebilly/php please re-review

**Review by justnero — APPROVED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**